ompi/mca/coll/ucx/coll_ucx_op.c

```diff
@@ -0,0 +1,465 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2011.  ALL RIGHTS RESERVED.
+ * Copyright (c) 2016      The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2019      Huawei Technologies Co., Ltd. All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "coll_ucx.h"
+#include "coll_ucx_request.h"
+
+#include "ompi/message/message.h"
+#include <inttypes.h>
+
+int mca_coll_ucx_start(size_t count, ompi_request_t** requests)
+{
+    mca_coll_ucx_persistent_op_t *preq=NULL;
+    ompi_request_t *tmp_req=NULL;
+    size_t i;
+
+    for (i = 0; i < count; ++i) {
+        preq = (mca_coll_ucx_persistent_op_t *)requests[i];
+        if ((preq == NULL) || (OMPI_REQUEST_COLL != preq->ompi.req_type)) {
+            /* Skip irrelevant requests */
+            continue;
+        }
+
+        COLL_UCX_ASSERT(preq->ompi.req_state != OMPI_REQUEST_INVALID);
+        preq->ompi.req_state = OMPI_REQUEST_ACTIVE;
+        mca_coll_ucx_request_reset(&preq->ompi);
+
+        tmp_req = ucg_collective_start_nb(preq->coll_desc);
+        if (tmp_req == NULL) {
+            COLL_UCX_VERBOSE(8, "collective completed immediately, completing persistent request %p",
```

Comment:
Is this in the critical code path?  If so, you might want to make sure that COLL_UCX_VERBOSE has negligible impact when disabled (e.g., is it wholly evaluated in the macro, maybe with an OPAL_UNLIKELY?).